### ISSUE TITLE
[URGENT] Degenerate derivatives

### ISSUE THREAD

**Issue by oesteban** [poldracklab/fmriprep#1075]
Since #1001 was merged in, the bold derivatives (at least the `space-MNI...` ones) are broken.

My intuition is that these changes on the Datasink introduced some problem https://github.com/poldracklab/fmriprep/compare/744de95e8c21...677c55647547#diff-7bf7d3ca48aa501c95ee753280fb3ee1

If I track back the files to the node that feeds the datasink, they look okay.

**Comment by effigies** [comment]
It's this bit:

https://github.com/poldracklab/fmriprep/blob/c76acd4efb89918fce517b3845688068705b3aa5/fmriprep/interfaces/bids.py#L252-L255

It assumes that the input is uncompressed.

**Comment by oesteban** [comment]
@mgxd can you/would you like to submit a fix?

**Comment by effigies** [comment]
I think if we do `ext.endswith('.gz') and not fname.endswith('.gz')`, that would work, but I guess we should actually make a smarter copy, like:

```Python
def copy_any(src, dst):
    src_gz = src.endswith('.gz')
    dst_gz = dst.endswith('.gz')
    if src_gz == dst_gz:
        copy(src, dst)
    src_open = gzip.open if src_gz else open
    dst_open = gzip.open if dst_gz else open
    with src_open(src, 'rb') as f_in:
        with dst_open(dst, 'wb') as f_out:
            copyfileobj(f_in, f_out)
```

And if we want to permit hardlinks, nipype's `utils.filemanip.copyfile` would be a good alternative to `shutil.copy`.

**Comment by oesteban** [comment]
Closed via #1077

**Comment by oesteban** [comment]
Some of the reportlets are broken since we merged #1077:

https://2844-53175327-gh.circle-artifacts.com/0/tmp/ds000210/derivatives/fmriprep/sub-02.html
